src/zope/schema/_bootstrapfields.py

```diff
@@ -530,6 +535,8 @@ def fromUnicode(self, str):
         zope.schema._bootstrapinterfaces.ConstraintNotSatisfied: (u'foo spam', '')
         """
         self.validate(str)
+        if self.unicode_normalization:
```

Comment:
It seems like normalization should happen before validation. The final value that's going to be stored is going to be normalized, so it seems like validation should gets its say about that final value.

Then it would seem like a test case that verifies that validation sees the final normalized data is needed.

Reply:
I implemented that and had to fix tests because when the new method was called first it failed before validation had a chance to fail like the test would expect.